### PR TITLE
perf: reduce concatenation in relative_path

### PR DESCRIPTION
**Pull request by jbedard**
The `bazel_skylib` `paths.join` contains a lot of logic unnecessary in this case that degrades performance.

See https://github.com/bazelbuild/bazel-skylib/blob/1.7.1/lib/paths.bzl#L90-L98

Also removed duplicate `len(..)` calls and removed the use of `zip` and `basename` which both construct a new array.

---

### Test plan

- Covered by existing test cases

### CODE REVIEW

**Comment by DavidZbarsky-at** [comment]
Before:
image


After:
image

**Comment by DavidZbarsky-at** [comment]
image

File: lib/private/paths.bzl [hunk 1]
```diff
@@ -29,33 +29,30 @@ def _relative_file(to_file, frm_file):
...
     split_point = 1
 
     # if either of the paths starts with ../ then assume that any shared paths are a coincidence
-    if to_segments[0] != ".." and frm_segments != "..":
-        longest_common = []
-        for to_seg, frm_seg in zip(to_segments, frm_segments):
-            if to_seg == frm_seg:
-                longest_common.append(to_seg)
-            else:
+    if to_segments[0] != ".." and frm_segments[0] != "..":
```
Comment (jbedard) [inline lib/private/paths.bzl:45]: Was the missing `[0]` on `frm_segments` a bug? Is there a test that should have caught that?

**Comment by DavidZbarsky-at** [comment]
Seems about the same as the previous one
image
image

File: lib/private/paths.bzl (hunk 1 above)
Comment (gregmagolan) [reply lib/private/paths.bzl:45]: There are tests but guess none caught this case

File: lib/private/paths.bzl [hunk 2]
```diff
@@ -127,7 +126,7 @@ def _to_repository_relative_path(file):
     """
 
     if file.short_path.startswith("../"):
-        return "/".join(file.short_path.split("/")[2:])
+        return file.short_path[file.short_path.find("/", 3) + 1:]
```
Comment (gregmagolan) [inline lib/private/paths.bzl:129]: clever with the start of the search pattern at 3 👍

File: lib/private/paths.bzl (hunk 1 above)
Comment (gregmagolan) [reply lib/private/paths.bzl:45]: Test cases are in `lib/tests/paths_test.bzl`

**Review by gregmagolan — APPROVED** [review]
🦖

**Pull request merged**